lib/Driver/ToolChains.cpp

```diff
@@ -443,71 +443,79 @@ ToolChain::constructInvocation(const CompileJobAction &job,
   Arguments.push_back("-module-name");
   Arguments.push_back(context.Args.MakeArgString(context.OI.ModuleName));
 
-  auto ModuleOutputPath =
-    context.Output.getAdditionalOutputForType(types::ID::TY_SwiftModuleFile);
-  if (!ModuleOutputPath.empty()) {
-    Arguments.push_back("-emit-module-path");
-    Arguments.push_back(context.Args.MakeArgString(ModuleOutputPath));
+  for (auto ModuleOutputPath :
+         context.Output.getAdditionalOutputsForType(types::ID::TY_SwiftModuleFile)) {
+    if (!ModuleOutputPath.empty()) {
+      Arguments.push_back("-emit-module-path");
+      Arguments.push_back(context.Args.MakeArgString(ModuleOutputPath));
+    }
   }
 
-  auto ObjCHeaderOutputPath =
-    context.Output.getAdditionalOutputForType(types::ID::TY_ObjCHeader);
-  if (!ObjCHeaderOutputPath.empty()) {
-    assert(context.OI.CompilerMode == OutputInfo::Mode::SingleCompile &&
-           "The Swift tool should only emit an Obj-C header in single compile"
-           "mode!");
+  for (auto ObjCHeaderOutputPath :
+         context.Output.getAdditionalOutputsForType(types::ID::TY_ObjCHeader)) {
+    if (!ObjCHeaderOutputPath.empty()) {
```

Comment:
This condition no longer makes sense; it should be asserting that the whole vector is empty.

Reply:
And come to think of it, it's not necessary for any of the others either.

Reply:
Hm. We don't assert non-empty names, but .. eek, we _are_ returning a nonempty vector when the names aren't present. I'll put a non-emptiness assertion in too and remove the check, since nothing good can possibly happen with empty names.